tests/integration/test_opensearch.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# Copyright 2025 Ubuntu
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import jubilant
+import yaml
+from helpers import OPENSEARCH_MODEL_CONFIG, get_application_data
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+KUBEFLOW_INTEGRATOR_APP_NAME = "kubeflow-integrator"
+OPENSEARCH_RELATION_NAME = "opensearch"
+
+OPENSEARCH_CHARM = "opensearch"
+OPENSEARCH_CHANNEL = "2/edge"
+OPENSEARCH_PROFILE_CONFIG = "testing"
+OPENSEARCH_APP_NAME = "opensearch"
+
+SELF_SIGNED_CERTIFICATES_CHARM = "self-signed-certificates"
+SELF_SIGNED_CERTIFICATES_CHANNEL = "latest/stable"
+SELF_SIGNED_CERTIFICATES_APP_NAME = "self-signed-certificates"
+
+
+def test_deploy_and_config_opensearch(juju: jubilant.Juju, kubeflow_integrator: str):
+    """Deploy Kubeflow Integrator, configure opensearch-index-name. The charm should be in a blocked status waiting to be integrated with opensearch."""
+    logger.info("Deploying Kubeflow Integrator charm")
+    # When:
+    juju.deploy(
+        kubeflow_integrator, app=KUBEFLOW_INTEGRATOR_APP_NAME, config={"profile": "profile-name"}
+    )
+
+    # Assert:
+    juju.wait(
+        lambda status: jubilant.all_active(status) and jubilant.all_agents_idle(status), delay=5
+    )
+
+    logger.info("Configure `opensearch-index-name` config option")
+    # When:
+
+    juju.config(KUBEFLOW_INTEGRATOR_APP_NAME, {"opensearch-index-name": "index-name"})
+
+    # Assert:
+    status = juju.wait(
+        lambda status: jubilant.all_blocked(status) and jubilant.all_agents_idle(status), delay=5
+    )
+
+    # Assert:
+    assert (
+        "Charm waiting to be integrated with OpenSearch"
+        in status.apps[KUBEFLOW_INTEGRATOR_APP_NAME].app_status.message
+    )
+    assert (
+        "Charm waiting to be integrated with OpenSearch"
+        in status.apps[KUBEFLOW_INTEGRATOR_APP_NAME]
+        .units[f"{KUBEFLOW_INTEGRATOR_APP_NAME}/0"]
```

Comment:
we should perhaps restrict this status to either the application or unit, to stay consistent with the UX of the data integrator